server/src/test/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointActionTests.java

```diff
@@ -0,0 +1,159 @@
+
+public class PublishCheckpointActionTests extends OpenSearchTestCase {
+
+    private ThreadPool threadPool;
+    private CapturingTransport transport;
+    private ClusterService clusterService;
+    private TransportService transportService;
+    private ShardStateAction shardStateAction;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        threadPool = new TestThreadPool(getClass().getName());
+        transport = new CapturingTransport();
+        clusterService = createClusterService(threadPool);
+        transportService = transport.createTransportService(
+            clusterService.getSettings(),
+            threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            boundAddress -> clusterService.localNode(),
+            null,
+            Collections.emptySet()
+        );
+        transportService.start();
+        transportService.acceptIncomingRequests();
+        shardStateAction = new ShardStateAction(clusterService, transportService, null, null, threadPool);
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        try {
+            IOUtils.close(transportService, clusterService, transport);
+        } finally {
+            terminate(threadPool);
+        }
+        super.tearDown();
+    }
+
+    public void testPublishCheckpointActionOnPrimary() throws InterruptedException{
+        final IndicesService indicesService = mock(IndicesService.class);
+
+        final Index index = new Index("index", "uuid");
+        final IndexService indexService = mock(IndexService.class);
+        when(indicesService.indexServiceSafe(index)).thenReturn(indexService);
+
+        final int id = randomIntBetween(0, 4);
+        final IndexShard indexShard = mock(IndexShard.class);
+        when(indexService.getShard(id)).thenReturn(indexShard);
+
+        final ShardId shardId = new ShardId(index, id);
+        when(indexShard.shardId()).thenReturn(shardId);
+
+        final RecoverySettings recoverySettings = new RecoverySettings(Settings.EMPTY, clusterService.getClusterSettings());
+
+        final PublishCheckpointAction action = new PublishCheckpointAction(
+            Settings.EMPTY,
+            transportService,
+            clusterService,
+            indicesService,
+            threadPool,
+            shardStateAction,
+            new ActionFilters(Collections.emptySet())
+        );
+
+        final ReplicationCheckpoint checkpoint = new ReplicationCheckpoint(indexShard.shardId(), 1111, 111, 11, 1);
+
+        final PublishCheckpointRequest request = new PublishCheckpointRequest(checkpoint);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        action.shardOperationOnPrimary(request, indexShard, new LatchedActionListener<>(ActionTestUtils.assertNoFailureListener(result -> {
+            // we should forward the request containing the current retention leases to the replica
```

Comment:
This looks leftover from a test specific to retention leases?

Reply:
Yes, sorry. I will remove it in next commit